File: 1-3-URLify.py
#  1.3 URLify

#  Write a method that replaces all spaces in a string with '%20'. 
# You may assume that the string has sufficient space at the end to hold the additional characters, and that you are given the length of the string.

def urlify(string):
    stripped = string.strip()
    url = ''
    for c in stripped:
        if c == ' ':
            url += '%20'
        else:
            url += c
    return url

print(urlify("Mr John Smith    "))  #  Mr%20John%20Smith

File: 1-3-URLify.js
//  1.3 URLify

//  Write a method to replace all spaces in a string with '%20'. 
// You may assume that the string has sufficient space at the end to hold the additional characters, and that you are given the "true" length of the string.
// (Note: If implementing in Java, please use a character array so that you an perform this operation in place.)

function URLify(s)
{
    let trimmed = s.trim();
    let url = trimmed.replace(" ", "%20");
    return url;
}

console.log(URLify(' hello world '));